File: lib_internet/src/main/java/world/share/lib_internet/bean/BaseResponse.java
package world.share.lib_internet.bean;

/**
 * @author mac
 */
public class BaseResponse<T> {
    private int code;
    private String message;
    private T result;

    public BaseResponse(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public BaseResponse(String message) {
        this.message = message;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public T getResult() {
        return result;
    }

    public void setResult(T result) {
        this.result = result;
    }

    public boolean isOk() {
        return code == 200;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}

File: lib_base/src/main/java/world/share/lib_base/internet/bean/BaseResponse.kt
package world.share.lib_base.internet.bean

/**
 * @author mac
 */
class BaseResponse<T> {
    var code = 0
    var message: String
    var result: T? = null
        private set

    constructor(code: Int, message: String) {
        this.code = code
        this.message = message
    }

    constructor(message: String) {
        this.message = message
    }

    fun setResult(result: T) {
        this.result = result
    }

    val isOk: Boolean
        get() = code == 200

}